polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -552,21 +552,13 @@ Value MLIRScanner::castToMemSpace(Value Val, unsigned MemSpace) {
         if (ValType.getMemorySpaceAsInt() == MemSpace)
           return Val;
 
-        Value NewVal = Builder.create<polygeist::Memref2PointerOp>(
-            Loc,
-            LLVM::LLVMPointerType::get(ValType.getElementType(),
-                                       ValType.getMemorySpaceAsInt()),
-            Val);
-        NewVal = Builder.create<LLVM::AddrSpaceCastOp>(
-            Loc, LLVM::LLVMPointerType::get(ValType.getElementType(), MemSpace),
-            NewVal);
-        return Builder.create<polygeist::Pointer2MemrefOp>(
+        return Builder.create<sycl::SYCLAddrSpaceCastOp>(
```

Comment:
Can we generate https://mlir.llvm.org/docs/Dialects/MemRef/#memrefmemory_space_cast-mlirmemrefmemoryspacecastop ?

Reply:
Added TODO, it is not part of intel/llvm, so cannot use yet.